src/components/file-input/file-input.style.tsx

```diff
@@ -0,0 +1,91 @@
+import styled, { css } from "styled-components";
+import type { InputProps } from "../../__internal__/input";
+import type { ValidationProps } from "../../__internal__/validations";
+import StyledTypography from "../typography/typography.style";
+
+export const StyledHintText = styled.div`
+  margin-top: 0px;
+  margin-bottom: 8px;
+  color: var(--colorsUtilityYin055);
+  font-size: 14px;
+`;
+
+export const ErrorBorder = styled.span`
+  position: absolute;
+  z-index: 6;
+  width: 2px;
+  background-color: var(--colorsSemanticNegative500);
+  left: -12px;
+  bottom: 0px;
+  top: 0px;
+`;
+
+export const StyledHiddenFileInput = styled.input<InputProps>`
+  display: none;
+`;
+
+interface StyledFileInputPresentationProps
+  extends Pick<ValidationProps, "error"> {
+  isDisabled?: boolean;
+  isDraggedOver?: boolean;
+  isDraggingFile?: boolean;
+  hasUploadStatus?: boolean;
+  maxHeight?: string;
+  maxWidth?: string;
+  minHeight?: string;
+  minWidth?: string;
+}
+
+export const StyledFileInputPresentation = styled.div<StyledFileInputPresentationProps>`
+  ${({ hasUploadStatus, minWidth, minHeight, maxWidth, maxHeight }) => css`
+    min-width: ${minWidth};
+    min-height: ${minHeight};
+    max-width: ${maxWidth};
+    ${!hasUploadStatus &&
+    css`
+      padding: 12px;
+      max-height: ${maxHeight};
+    `}
+  `}
+
+  ${({ hasUploadStatus, isDisabled, isDraggedOver, isDraggingFile, error }) => {
+    const borderWidth =
+      error || (isDraggingFile && !isDisabled) ? "2px" : "1px";
+    let borderColorToken = "colorsUtilityMajor300";
+    let backgroundColorToken = "colorsUtilityYang100";
+    if (isDisabled) {
+      borderColorToken = "colorsUtilityDisabled600";
+      backgroundColorToken = "colorsUtilityDisabled400";
+    } else {
+      if (isDraggedOver) {
+        borderColorToken = "colorsUtilityMajor400";
+        backgroundColorToken = "colorsUtilityMajor100";
+      }
+      if (error) {
+        borderColorToken = "colorsSemanticNegative600";
+      }
+    }
+    return (
+      !hasUploadStatus &&
+      css`
+        display: flex;
+        flex-direction: column;
+        flex-wrap: wrap;
+        justify-content: center;
+        align-content: center;
+        align-items: center;
+        text-align: center;
+        gap: 12px;
+        border-radius: 4px;
```

Comment:
question: could this be a design token?

I'm just wondering if it's possible that there's consistency here that DS want standard rounded corners across components. It's more in case they want to square off corners again it would be easier with a design token... But maybe I'm just worrying about something that won't happen

Reply:
Good question. I grabbed the ones I could from Figma but mostly it's only colours that that seems to show tokens for. I've now had a catch up with Laia from the DS team and she's pointed out a few tokens I missed (in terms of the colour etc actually being wrong, I mean) and has sent me a full list of them all - so absolutely any styles that come from tokens will be done properly.

Reply:
in fact, similar to Ed's suggestion above with the StyledHintText, I'm just going to remove this entirely and inport this styled component from `textbox.style`, as several components already do (and pending a later tidy-up that moves these to a common file).